templates/indicators/indicator_form_common_js.html

```diff
@@ -1014,6 +1014,39 @@
         }
     }
 
+    /*
+     * Higher-order function to automate validation internals for a length-validated textarea
+     * Takes the field name (i.e. "justification") and max length (i.e. 400) and returns a validator for
+     * that field, which when called adds invalid classes/markings/labels, and returns false if not valid
+     * called both on the keyup event listeners below, and as part of whole-form validation
+     */
+    function getTextAreaLengthValidator(fieldName, length) {
```

Comment:
This is a textbook example of a closure.

Reply:
and I got there the way I always do - writing the loop without a closure and wondering why everything is validating the last field in the list...